lib/shared/src/datetime.rs

```diff
@@ -0,0 +1,76 @@
+use chrono::{DateTime, Local, ParseError, TimeZone as _, Utc};
+use chrono_tz::Tz;
+use derivative::Derivative;
+use std::fmt::{self, Debug};
+
+#[derive(Clone, Copy, Debug, Derivative)]
+#[derivative(Default)]
+pub enum TimeZone {
+    #[derivative(Default)]
+    Local,
+    Named(Tz),
+}
+
+/// This is a wrapper trait to allow `TimeZone` types to be passed genericly.
+impl TimeZone {
+    pub(super) fn datetime_from_str(
+        &self,
+        s: &str,
+        format: &str,
+    ) -> Result<DateTime<Utc>, ParseError> {
+        match self {
+            Self::Local => Local.datetime_from_str(s, format).map(datetime_to_utc),
+            Self::Named(tz) => tz.datetime_from_str(s, format).map(datetime_to_utc),
+        }
+    }
+
+    pub fn parse(s: &str) -> Option<Self> {
+        match s {
+            "" | "local" => Some(Self::Local),
+            _ => s.parse::<Tz>().ok().map(Self::Named),
```

Comment:
Nit: I might have usually seen this as defining some placeholder variable and using that in the match arm rather than referring to the outer scope. Is what you have here more common though?

Reply:
I don't know. I think I have seen the above more often than what you describe, but naming the variable in the arm is a newer feature.

Reply:
Gotcha, I was mostly just curious.